ring.go

```diff
@@ -58,3 +58,15 @@ func (r *ring) avg() int {
 	}
 	return r.sum / len(r.data)
 }
+
+func (r *ring) sequentialData() []int {
+	index := r.idx
+	slice := make([]int, r.maxLen)
+	if len(r.data) < r.maxLen {
+		copy(slice, r.data)
+		return slice
+	}
```

Comment:
```suggestion
	if index == 0 {
		copy(slice, r.data)
		return slice
	}
```
we'd better consider such common case.